
Microbial communities in the banana rhizosphere are varied and play essential roles in
promoting plant growth and increasing crop yield. Their abundance in soil habitats is a
potential indicator of soil health and quality. Despite the well-known benefits of rhizosphere
microorganisms in banana cultivation, their genomic and functional diversity remains largely
unexplored within smallholder agroecosystems. Here, we examined the community
composition and functional potentials of bacteria in the banana rhizospheric soils collected
from Kiambu, Kenya using Illumina Novaseq sequencing of 16S rRNA gene amplicons and
shotgun metagenomic profiles. Variations of soil physicochemical parameters across the
study sites were assessed. The parameters varied across the sites with soils in Gituamba and
Ngenda depicting better soil fertility characteristics than Mangu. Alpha diversity indicated
higher bacterial diversity in Gituamba than Mangu, with significant differences in observed
features and phylogenetic distance metrics. In contrast, a moderate diversity of the bacterial
community in Ngenda based on Shannon and Simpson indexes was observed. The dominant
phyla across the study sites were Proteobacteria, Actinobacteria and Cyanobacteria.
Predicted functional profiling of amplicon gene profiles showed that metabolic pathways
were significantly enriched in Gituamba than Mangu. Overall, the functional profiling
indicated that, predicted metabolic pathways across the study sites were linked to genes
encoded by the members of the most abundant bacterial phyla in the soil environments,
majorly contributing to beneficial roles for soil health and crop yield. This study offers
methods to reveal the banana rhizosphere as a rich reservoir for microorganisms of
agricultural and biotechnological significance, which can promote sustainable agriculture.

Keywords: Bacteria, banana rhizosphere, metagenomics, soil physicochemical properties,
sustainable agriculture.
